src/Nethermind/Nethermind.Trie/Pruning/TrieStore.cs

```diff
@@ -412,36 +409,34 @@ private bool CanPruneCacheFurther()
             {
                 if (_logger.IsDebug) _logger.Debug("Elevated pruning starting");
 
-                List<BlockCommitSet> toAddBack = new();
-
-                List<BlockCommitSet> candidateSets = new();
-                while (_commitSetQueue.TryDequeue(out BlockCommitSet? frontSet))
+                using ArrayPoolList<BlockCommitSet> toAddBack = new(_commitSetQueue.Count * 2);
```

Comment:
Why *2?

Reply:
To have headroom if the queue grows during pruning

Reply:
Also feel free to review ArrayPoolList code, though it is used in our code without issues already.

Reply:
Ok I will remove it